include/zephyr/pm/state.h

```diff
@@ -358,6 +364,34 @@ struct pm_state_constraint {
 			Z_PM_STATE_FROM_DT_CPU, (), node_id)		       \
 	}
 
+/**
+ * @brief initialize a device pm constraint with information from devicetree.
+ *
+ * @param node_id Node identifier.
+ */
+#define PM_STATE_CONSTRAINT_INIT(node_id)                                     \
+	{                                                                     \
+		.state = PM_STATE_DT_INIT(node_id),                           \
+		.substate_id = DT_PROP_OR(node_id, substate_id, 0),           \
+	}
+
+#define PM_STATE_CONSTRAINT_REF(node_id, phandle, idx)                        \
+	PM_STATE_CONSTRAINT_INIT(DT_PHANDLE_BY_IDX(node_id, phandle, idx))
+
+#define PM_STATE_CONSTRAINTS_LIST_NAME(node_id, phandles)                     \
+	node_id ## _ ## phandles ## _constraints
+
+#define PM_STATE_CONSTRAINTS_LIST_DEFINE(node_id, phandles)                                \
+	struct pm_state_constraint PM_STATE_CONSTRAINTS_LIST_NAME(node_id, phandles)[] =   \
+	{                                                                                  \
+		DT_FOREACH_PROP_ELEM_SEP(node_id, phandles, PM_STATE_CONSTRAINT_REF, (,))  \
+	}
+
+#define PM_STATE_CONSTRAINTS_LIST_GET(node_id, phandles)                                   \
```

Comment:
Following the previous note about the naming, `PM_STATE_CONSTRAINTS_GET` would be nicer IMO